### PR TITLE
[HOPSWORKS-1467] min provenance migrate/rollback (idempotent)

### PR DESCRIPTION
**Pull request by o-alex**

### CODE REVIEW

File: src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/HopsClient.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+    
+    Path hdfsPath = new Path(hdfsConfFile.getAbsolutePath());
+    Path hadoopPath = new Path(hadoopConfFile.getAbsolutePath());
+    
+    Configuration conf = new Configuration();
+    conf.addResource(hadoopPath);
+    conf.addResource(hdfsPath);
+    conf.set(CommonConfigurationKeys.FS_PERMISSIONS_UMASK_KEY, "0027");
+    conf.setStrings("dfs.namenode.rpc-address", getRPCEndpoint(connection));
```
Comment (SirOibaf) [inline src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/HopsClient.java]: Why do you need to query the database to know the rpc addresses of the namenodes when you just loaded both `hdfs-site` and `core-site`?

File: pom.xml [hunk 2]
```diff
@@ -92,6 +92,39 @@
             <version>2.6.2</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.persistence</groupId>
+            <artifactId>eclipselink</artifactId>
+            <version>2.5.2</version>
+            <type>jar</type>
```
Comment (SirOibaf) [inline pom.xml]: Set this as provided.

File: src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/UpdateProvenance.java [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+      while (allProjectsResultSet.next()) {
+        //get project inode
+        projectInodeStmt = getProjectInodeStmt(allProjectsResultSet);
+        ResultSet projectInodeResultSet = projectInodeStmt.executeQuery();
+        if(!projectInodeResultSet.next()) {
+          throw new IllegalStateException("project inode not found");
+        }
+        
+        ProjectParams projectParams = ProjectParams.instance(allProjectsResultSet, projectInodeResultSet);
+        LOGGER.debug("processing project:{}", projectParams.projectName);
```
Comment (SirOibaf) [inline src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/UpdateProvenance.java]: Can you set the project logs to info so we can see how it's progressing?

File: pom.xml (hunk 2 above)
Comment (o-alex) [reply pom.xml]: fixed

File: src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/UpdateProvenance.java (hunk 3 above)
Comment (o-alex) [reply src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/UpdateProvenance.java]: fixed

File: src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/HopsClient.java (hunk 1 above)
Comment (o-alex) [reply src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/HopsClient.java]: i adapted this from one of the hops tests... are you sure I don't need to set the rpc address?

Comment (o-alex) [reply src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/HopsClient.java]: fixed

**Pull request merged**